mobilecoind-dev-faucet/src/lib.rs

```diff
@@ -0,0 +1,296 @@
+            monitor_public_address.clone(),
+            minimum_fees,
+            faucet_payout_amounts.clone(),
+            config.target_queue_depth,
+            Duration::from_millis(config.worker_poll_period_ms),
+            logger,
+        );
+
+        let logger = logger.new(o!("thread" => "http"));
+
+        Ok(State {
+            mobilecoind_api_client,
+            account_key,
+            monitor_id,
+            monitor_b58_address,
+            faucet_payout_amounts,
+            worker,
+            logger,
+        })
+    }
+
+    /// Handle a "post" to the faucet, which requests a payment from the faucet.
+    /// Returns either the mobilecoind success response or an error string.
+    pub async fn handle_post(
+        &self,
+        req: &JsonFaucetRequest,
+    ) -> Result<mc_mobilecoind_api::SubmitTxResponse, String> {
+        let printable_wrapper = PrintableWrapper::b58_decode(req.b58_address.clone())
+            .map_err(|err| format!("Could not decode b58 address: {}", err))?;
+
+        let public_address = if printable_wrapper.has_public_address() {
+            printable_wrapper.get_public_address()
+        } else {
+            return Err(format!(
+                "b58 address '{}' is not a public address",
+                req.b58_address
+            ));
+        };
+
+        let token_id = TokenId::from(req.token_id.unwrap_or_default().as_ref());
+
+        let utxo_record = self.worker.get_utxo(token_id)?;
+        log::trace!(
+            self.logger,
+            "Got a UTXO: key_image = {:?}, value = {}",
+            KeyImage::try_from(utxo_record.utxo.get_key_image()).unwrap(),
+            utxo_record.utxo.value
+        );
+
+        // Generate a Tx sending this specific TxOut, less fees
+        let mut req = mc_mobilecoind_api::GenerateTxFromTxOutListRequest::new();
+        req.set_account_key((&self.account_key).into());
+        req.set_input_list(vec![utxo_record.utxo].into());
+        req.set_receiver(public_address.clone());
+        req.set_token_id(*token_id);
+
+        let resp = self
+            .mobilecoind_api_client
+            .generate_tx_from_tx_out_list_async(&req)
+            .map_err(|err| format!("Failed to build Tx: {}", err))?
+            .await
+            .map_err(|err| format!("Build Tx ended in error: {}", err))?;
+
+        // Submit the tx proposal
+        let mut req = mc_mobilecoind_api::SubmitTxRequest::new();
+        req.set_tx_proposal(resp.get_tx_proposal().clone());
+
+        let resp = self
+            .mobilecoind_api_client
+            .submit_tx_async(&req)
+            .map_err(|err| format!("Failed to submit Tx: {}", err))?
+            .await
+            .map_err(|err| format!("Submit Tx ended in error: {}", err))?;
+
+        // Tell the worker that this utxo was submitted, so that it can track and
+        // recycle the utxo if this payment fails
+        let _ = utxo_record.sender.send(resp.clone());
```

Comment:
hmm, so this error occurs if the one-shot receiver was dropped by the worker thread before we sent this.

i feel like, this represents some kind of internal error or race, but we still submitted a transaction successfully. if we propagate this error, then we are telling the faucet user that the tx failed, but that may not be true -- at this point I think we submitted something successfully but had an internal error passing a message.

i feel like the right thing to do is log but not propagate, because i think the right thing for the faucet to do is return `success: true` and let the client track what happens the payment if they want to